### ISSUE TITLE
recursive scp

### ISSUE THREAD

**Issue by cyrillef** [fioan89/sourcesync#23]
Hi,

In your documentation:
If SCP protocol is used, please take note that this type of connection can only sync the file, it cannot create the entire directory tree where the file will be saved. Thus you need to have the tree created. This is not a bug of sourcesync. SCP is not a file protocol like FTP. It's only purpose is to transfer files.

but if you do:
scp -r . root@192.168.1.54:/root

that will copy all files and directory structure even if it is not yet created on the remote machine.

**Comment by curunoir** [comment]
I agree, would be a great feature like the old sftp eclipse plugin did

**Comment by mohamed-ezz** [comment]
+1. Syncing a selected directory is much needed

**Comment by fioan89** [comment]
Hi @cyrillef @curunoir @mohamed-ezz . I just wanted to let you know that I've released Sourcesync Pro, a paid variant for Sourcesync with extra features. One of them is recursive, folder transfer for SCP. Among other things, Sourcesync Pro also adds support auto-sync when files are saved and local vs remote file comparison (integrated with IntelliJ Diff tool). Please check out https://plugins.jetbrains.com/plugin/22318-source-synchronizer-pro

I will continue to maintain and improve Sourcesync - the free&opensource version however some features that are more complex to implement and maintain will live in the Pro variant. Thank you for your understanding and support.

Warm regards,
Ioan
